xwiki-platform-core/xwiki-platform-eventstream/src/main/java/org/xwiki/eventstream/RecordableEventDescriptorContainer.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.eventstream;
+
+import java.util.List;
+
+import org.xwiki.component.annotation.Role;
+import org.xwiki.stability.Unstable;
+
+/**
+ * Provide a container for {@link RecordableEventDescriptor} that can’t be registered through the
+ * component manager.
```

Comment:
This makes me ask 2 questions:

1. Since this "container" does not contain **all** descriptors, we should think about an other name that reflect that. It could be "PageNotificationDescriptorContainer" or "PageNotificationDescriptorManager".
2. Since this role has been created for the sake of the implementation, it might be better to put it in the "internal" package. Otherwise it means extensions developers can use this API, and I don't think we want it.